third_party/ijar/test/zip_test.sh

```diff
@@ -188,4 +188,26 @@ function test_zipper_specify_path() {
       || fail "Unzip after zipper output is not expected"
 }
 
+function test_zipper_permissions() {
+  #(rm -fr ${TEST_TMPDIR}/* || true)
+  mkdir -p ${TEST_TMPDIR}/files
+  printf "#!/bin/sh\nexit 0\n" > ${TEST_TMPDIR}/files/executable
+  printf "#!/bin/sh\nexit 0\n" > ${TEST_TMPDIR}/files/non_executable
+  chmod +x ${TEST_TMPDIR}/files/executable
+  chmod -x ${TEST_TMPDIR}/files/non_executable
+
+  ${ZIPPER} cC ${TEST_TMPDIR}/output.zip \
+      executable=${TEST_TMPDIR}/files/executable \
+      non_executable=${TEST_TMPDIR}/files/non_executable
+
+  mkdir -p ${TEST_TMPDIR}/out
+  (cd ${TEST_TMPDIR}/out && $UNZIP -q ${TEST_TMPDIR}/output.zip)
+
+  ${TEST_TMPDIR}/out/executable || fail "out/executable should have been executable"
+
+  trap 'return 0' ERR
+  ${TEST_TMPDIR}/out/non_executable
```

Comment:
How about writing this as '...non_executable && fail "foo bar"'?

Reply:
Won't that still cause the test to fail?
`non_executable` will fail due to a permissions error so the overall test will fail (even though the `fail "blah") doesn't get invoked.

That being said, it looks like there's something wrong with this test anyway on the macOS platform. I don't have a mac to test with so I'm not sure what to do there :/

(I have no idea how to write `bash` tests - let alone one that's expecting a permissions error)

Reply:
(Actually, I'm not sure about the buildkite stuff failing on macOS - the warning seems to have gone away?)